### PR TITLE
L4 NSIDC CDR ice concentration converter

### LINKED ISSUE #1080: converter for L4 sea ice concentration from NSIDC CDR
## Description
Create a converter for L4 sea ice concentration from NSIDC CDR

## Acceptance Criteria (Definition of Done)
Pass ctest

### PR DESCRIPTION
**Pull request by ShastriPaturi**
## Description
This PR creates an IODA converter for sea-ice concentration from L4 NSIDC CDR data. 
These data can be used for diagnostic purposes but are also assimilated by CPC for the NG-GODAS real-time runs.

### Issue(s) addressed

Link the issues to be closed with this PR
- fixes #1080 

## Acceptance Criteria (Definition of Done)
Pass ctest

### CODE REVIEW

File: src/marine/nsidc_l4cdr_ice2ioda.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+        data_in = {}
+        input_vars = (
+            'stdev_of_cdr_seaice_conc',
+            'cdr_seaice_conc',
+            'longitude',
+            'latitude')
+
+        # get the QC flags, and calculate a mask from the non-missing values
+        # (since L3 files are mostly empty, fields need a mask applied immediately
```
Comment (kbhargava) [inline src/marine/nsidc_l4cdr_ice2ioda.py:53]: Stupid question: these are L4 so do you still need to do filtering here?

Comment (kbhargava) [reply src/marine/nsidc_l4cdr_ice2ioda.py]: Ahh!! Sorry Friday evening ..my brain isn;t working anymore :)

**Review by mjhossen — APPROVED** [review]
ctest is passed with both norm and conversion

**Review by kbhargava — APPROVED** [review]
Buillds and ctests pass on orion using spack stack,

**Review by srherbener — APPROVED** [review]
Thanks for adding this converter and putting in a new test as well!

**Pull request merged**